Update/homepage improvements

breezyfasano opened the pull request:
## What are you changing in this pull request and why?
Changes as outlined in the [Asana task](https://app.asana.com/0/1200099998847559/1203986472821200/f)

- [x]  When announcement bar is not present, make sure diamonds in hero do not overlap the header
- [x]  Make the spacing between the rows of cards is an even 2rem across the board  
- [x]  Card fonts - font size is 20, line height is 30
    - [x]  Also for “Latest from Developer Blog”
- [x]  Icon width is 25px
- [x]  50px padding on left and right of entire page.
- [x]  Latest from Dev Blog Section
    - [x]  Reduce space between each card - 30px
    - [x] Transparent gradient on sides of swiper container

breezyfasano committed:
Font size and spacing changes

breezyfasano committed:
Spacing on left and right 50px

breezyfasano committed:
Add inset shadow to swiper

breezyfasano committed:
Merge branch 'current' of github.com:dbt-labs/docs.getdbt.com into update/homepage-improvements

wendyvasq commented on website/src/css/custom.css:
```diff
@@ -1722,8 +1722,9 @@ section > h2:not(.resource-section) {
-  margin-bottom: 0;
+  padding-bottom: 0;
+  margin-bottom: 6.25rem;
+  box-shadow: inset 30px 0 30px -30px rgba(0, 0, 0, 0.3), inset -30px 0 30px -30px rgba(0, 0, 0, 0.3);
```
I never seen this before so I am not that confident if this is the right  for css.   
The box shadowing works when I tested it. Welp,  I learned something new today.

john-rock replied:
On light mode it looks like that gradient is a bit harsh at the edge. Maybe we should smooth that out a bit.

![dbt_Developer_Hub](https://user-images.githubusercontent.com/46692803/225917446-556bf962-8463-43ce-8cad-218377a8c3e3.png)

lfracalossiFA replied:
@wendyvasq For informational purposes... there are some great tools out there that will take inputs from sliders and give you the `box-shadow` values. This is the one I'm familiar with: https://cssgenerator.org/box-shadow-css-generator.html

JKarlavige replied:
Yeah that's a good callout @lfracalossiFA, i've always used tools to generate the css for box shadows. Maybe @breezyfasano is a wizard and can whip these together by hand, but i definitely cannot 😅 

Along with using tools, you can also get the box-shadow css from Figma mockups. If you click an element in Figma that has a box-shadow, then click `Inspect` in the upper right, it gives you all the css for that element, including the box-shadow!
<img width="232" alt="image" src="https://user-images.githubusercontent.com/13647110/225924160-29ec3612-080f-4857-8c31-1fc58f6c9d52.png">

lfracalossiFA replied:
Also I'm seeing the same think in dark mode:

<img width="363" alt="Screenshot 2023-03-17 at 9 58 34 AM" src="https://user-images.githubusercontent.com/72401230/225926233-e5393685-4b02-4817-b6ad-ca6d9c099039.png">

I noticed you said you added a transparent gradient on the Swiper carousel, so I'm wondering if this is intentional? I'm not sure I like the look of it, though.

lfracalossiFA replied:
@JKarlavige Oh yeah, I was going to mention getting the style from the Figma comp, but then I realized there wasn't an actual Figma comp for this (I don't think?)

wendyvasq reviewed (dismissed):
lgtm !  I left a comment on a line of css for box shadowing that I am not too sure about.

john-rock reviewed (changes requested):
All looks good just had one callout on the Swiper gradient.

edit: Just realized Emily had called that out on the task.

lfracalossiFA commented:
Hey @breezyfasano , question -- I saw this in your notes:

> Card fonts - font size is 20, line height is 30
> Also for “Latest from Developer Blog”

"Latest from Developer Blog" looks to be 25px/41px. Is this correct?

lfracalossiFA commented on website/src/components/hero/styles.module.css:
```diff
@@ -29,7 +29,7 @@
     position: absolute;
     left: -50px;
-    top:150px;
+    top:125px;
```
Would it make sense to make this a `vh` value? It might scale a bit better on different screen sizes. I haven't tested this, though, so just a theory. Feel free to ignore me if not relevant :)

lfracalossiFA commented on website/src/components/postCarousel/index.js:
```diff
@@ -1,27 +1,29 @@
 import { Swiper, SwiperSlide } from 'swiper/react';
-import { Navigation } from 'swiper';
 import 'swiper/css';
+import { Navigation } from 'swiper';
```
For my own edification: any reason for moving this import down below the CSS?

breezyfasano replied:
There was no particular reason for this that I had in mind, to be honest!

breezyfasano replied:
I think I had tried to use a piece of the Swiper package for the shadow, but it didn't work out - so I reverted everything back. But it looked like that shook out to looking like I moved this down one line for no reason 😆

lfracalossiFA commented on website/src/components/postCarousel/index.js:
```diff
@@ -1,27 +1,29 @@
     <div>
       <Swiper
-        spaceBetween={60}
+        spaceBetween={30}
```
Are these values pixels?

breezyfasano replied:
Sure are!

lfracalossiFA commented on website/src/css/custom.css:
```diff
@@ -1722,8 +1722,9 @@ section > h2:not(.resource-section) {
+  box-shadow: inset 30px 0 30px -30px rgba(0, 0, 0, 0.3), inset -30px 0 30px -30px rgba(0, 0, 0, 0.3);
 }
 
 .home .swiper .swiper-button-next, .home .swiper .swiper-button-prev {
```
Curious why GitHub is showing me this line but not the rest of the file, when it looks like there were no more changes after this point... 🤔

lfracalossiFA commented:
And super-duper narrow on Galaxy S8+ (admittedly I'm not an Android user and I have no idea if anyone uses these any more, lol).

<img width="366" alt="Screenshot 2023-03-17 at 9 52 38 AM" src="https://user-images.githubusercontent.com/72401230/225924589-63530bc5-d1e6-4f7f-a3c7-d2b7839b6692.png">

lfracalossiFA reviewed (changes requested):
A few small things that I think should be changed! Would like Emily's read on the size of the carousel cards on mobile and if it should be fixed.

breezyfasano committed:
Remove shadow

breezyfasano commented:
Team - I went ahead and removed the shadow per Emily's request on the Asana thread!
Still adjusting working on the Swiper card widths on mobile per @lfracalossiFA's feedback.
Thanks for your comments, all!

breezyfasano committed:
Merge branch 'current' of github.com:dbt-labs/docs.getdbt.com into update/homepage-improvements

breezyfasano committed:
Center all h2s on mobile

breezyfasano committed:
Adjsut slider padding on mobile

breezyfasano commented:
@lfracalossiFA and @john-rock - re-requesting a review on this with some changes!
1. I've removed the gradient from the carousel re: Emily's comment in Asana
2. I've also adjusted the carousel on mobile to reflect a less scrunched view!

Let me know what y'all think & if this addressed your changes.
https://deploy-preview-3012--docs-getdbt-com.netlify.app/

lfracalossiFA reviewed (approved):
lgtm, but I do see eslint-check throwing up a bunch of warnings within the Files Changed tab. I don't know if those are super important, though.

JKarlavige commented:
@lfracalossiFA that's okay, the warnings have been in there. We just setup ESLint for PRs, and only sorted out the blocking errors so far, but could always go back and clean up the warnings at some point

breezyfasano committed:
Merge branch 'current' into update/homepage-improvements
